tests/ballerina-test/src/test/java/org/ballerinalang/test/auth/PermissionStoreTest.java

```diff
@@ -0,0 +1,125 @@
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+*  Unless required by applicable law or agreed to in writing,
+*  software distributed under the License is distributed on an
+*  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+*  KIND, either express or implied.  See the License for the
+*  specific language governing permissions and limitations
+*  under the License.
+*/
+
+package org.ballerinalang.test.auth;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.ballerinalang.config.ConfigRegistry;
+import org.ballerinalang.launcher.util.BCompileUtil;
+import org.ballerinalang.launcher.util.BRunUtil;
+import org.ballerinalang.launcher.util.CompileResult;
+import org.ballerinalang.model.values.BString;
+import org.ballerinalang.model.values.BStruct;
+import org.ballerinalang.model.values.BValue;
+import org.testng.Assert;
+import org.testng.annotations.AfterClass;
+import org.testng.annotations.BeforeClass;
+import org.testng.annotations.Test;
+
+import java.io.IOException;
+import java.nio.file.CopyOption;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.HashMap;
+import java.util.Map;
+
+import static java.nio.file.StandardCopyOption.REPLACE_EXISTING;
+
+public class PermissionStoreTest {
+    private static final Log log = LogFactory.getLog(PermissionStoreTest.class);
+    private static final String BALLERINA_CONF = "ballerina.conf";
+    private CompileResult compileResult;
+    private String resourceRoot;
+    private Path sourceRoot;
+    private Path ballerinaConfPath;
+    private Path ballerinaConfCopyPath;
+
+    @BeforeClass
+    public void setup() throws Exception {
+        resourceRoot = getClass().getProtectionDomain().getCodeSource().getLocation().getPath();
+        sourceRoot = Paths.get(resourceRoot, "test-src", "auth");
+        ballerinaConfPath = Paths
+                .get(resourceRoot, "datafiles", "config", "auth", "authorization", "permissionstore", BALLERINA_CONF);
+        ballerinaConfCopyPath = sourceRoot.resolve(BALLERINA_CONF);
+
+        // Copy the ballerina.conf to the source root before starting the tests
+        Files.copy(ballerinaConfPath, ballerinaConfCopyPath, new CopyOption[] { REPLACE_EXISTING });
+        compileResult = BCompileUtil.compile(sourceRoot.resolve("permission-store.bal").toString());
+
+        // load configs
+        ConfigRegistry registry = ConfigRegistry.getInstance();
+        registry.initRegistry(getRuntimeProperties(), ballerinaConfCopyPath);
+        registry.loadConfigurations();
+    }
+
+    private Map<String, String> getRuntimeProperties() {
+        Map<String, String> runtimeConfigs = new HashMap<>();
+        runtimeConfigs.put(BALLERINA_CONF,
+                Paths.get(resourceRoot, "datafiles", "config", "auth", "authorization", "permissionstore",
+                        BALLERINA_CONF).toString());
+        return runtimeConfigs;
+    }
+
+    @Test
```

Comment:
May be we can improve test case descriptions Eg:- @Test(description = "This test is added for.............")